hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/HddsServerUtil.java

```diff
@@ -472,31 +470,6 @@ public static SCMSecurityProtocolClientSideTranslatorPB getScmSecurityClient(
             ugi == null ? UserGroupInformation.getCurrentUser() : ugi));
   }
 
-  public static SCMSecurityProtocolClientSideTranslatorPB
```

Comment:
@fapifta , I didn't see obvious benefit of moving this getScmSecurityClientWithFixedDuration from HddsServerUtil.java to HASecurityUtils, could you explain it a bit? Changing this function location, cause a lot of changes in DefaultCertificateClient and HddsDatanodeService.

Reply:
This method was only referenced from HASecurityUtils#getRootCASignedSCMCert, hence the move and then I also made this private in HASecurityUtils.

I guess you are referring to getScmSecurityClientWithMaxRetry that is a change everywhere where we create a CertificateClient, as the creation of the ScmSecurityClient this way is moved out of the CertificateClient. By injecting the ScmSecurityClient via the constructor gives us the benefit of not having to be worried about initialization order in tests we do not need a setter just for testing, and we can easily inject mocking for the communicaiton via the constructor (actually we have to, which makes the test writer think about this, and do not just assume that the server communication would not be call from the test flow), also all the services can separately specify retries, timeouts and such, which can be a benefit in the future (but I admit it is not really a reason now).

Reply:
@fapifta , Yeah, I mean getScmSecurityClientWithMaxRetry, not getScmSecurityClientWithFixedDuration. To support every service can have different scm client retires, timeouts, we can just override the getScmSecureClient() function, I just did it in SCMCertificateClient:)

I still think it would better to handle the scm client creation in Certificate client internally, instead of now every time, it need create the scm client before instantiate a certificate client, avoiding many duplicate codes.

Reply:
Now I get the reason. We can get ConfigurationSource from SecurityConfig.getConfiguration(). So the DefaultCertificateClient constructor can just use the SecurityConfig.

Reply:
> We can get ConfigurationSource from SecurityConfig.getConfiguration().

After this change we cannot, this method is being removed.

Reply:
Okay, then I'm fine with the current solution.  

Thanks @fapifta for all the effort.  I will merge it.